lib/index.js

```diff
@@ -1,88 +1,152 @@
+'use strict';
+
 var AWS = require('aws-sdk');
-AWS.config.region = process.env.AWS_REGION;
 var ses = new AWS.SES();
-var path               = require('path');       // resolve template files
-var fs                 = require('fs');         // open tempalte files
-var Handlebars         = require('handlebars'); // compile templates
-var COMPILED_TEMPLATES = {};                    // cache compiled templates
+var path = require('path');             // -> resolve template files
+var fs = require('fs');                 // -> open template files
+var Handlebars = require('handlebars'); // -> compile templates
+var COMPILED_TEMPLATES = {};            // -> cache compiled templates
+
+AWS.config.region = process.env.AWS_REGION;
 
 /**
  * set_template_directory does exactly what its name suggests
  * @param {String} dir - the template directory
+ * @returns {undefined}
  */
-function set_template_directory(dir) {
-  if(!dir) {
-    throw "Please Set a Template Directory";
+function set_template_directory (dir) {
+  if (!dir) {
+    throw new Error('Please Set a Template Directory');
   }
-  var files = fs.readdirSync(dir); // this should be sync (on startup)
+  fs.readdirSync(dir);                  // this should be sync (on startup)
+
   process.env.TEMPLATE_DIRECTORY = dir; // set the env var
 }
 set_template_directory(process.env.TEMPLATE_DIRECTORY);
 
 /**
  * open template file sync (ONCE) and compile it!
  * @param {String} template_name - filename of template
- * @param {String} type - the file type
+ * @param {String} type          - the file type
+ * @returns {Object}             - compiled templates
  */
-function compile_template(template_name, type) {
+function compile_template (template_name, type) {
+  var filename = template_name + '.' + type;
+  var filepath = path.resolve(process.env.TEMPLATE_DIRECTORY, filename);
+  var template_cached = COMPILED_TEMPLATES[template_name + '.' + type];
+
+  var template = !template_cached ? fs.readFileSync(filepath, 'utf8') : '';
+  var compiled = Handlebars.compile(template);
```

Comment:
we could move the `Handlebars.compile(template)` from here to _inside_ the if statement on line 43. see: #74 (_not urgent but would nice-to-have_)